src/lint/lint.ts

```diff
@@ -68,12 +99,149 @@ export async function lint(options: Options, config: ProjectConfig) {
   const filtered = warnings.filter((w) => !filter.shouldIgnore(w));
 
   if (options.fix) {
-    return fix(filtered, options, config, analyzer, analysis);
+    const changedFiles = await fix(
+        filtered,
+        options,
+        config,
+        analyzer,
+        analysis,
+        editActionsToAlwaysApply);
+    if (watcher) {
+      // Some file watcher interfaces won't notice this change immediately after
+      // the one that initiated this lint run. Ensure that we notice these
+      // changes.
+      for (const changedFile of changedFiles) {
+        watcher.ensureChangeIsNoticed(changedFile);
+      }
+    }
+    if (changedFiles.size === 0 && options.reportIfNoFix) {
+      await report(filtered);
+    }
   } else {
     return report(filtered);
   }
 }
 
+async function watchLoop(
+    analyzer: Analyzer,
+    linter: lintLib.Linter,
+    options: Options,
+    config: ProjectConfig,
+    filter: WarningFilter) {
+  let analysis;
+  if (options.input) {
+    analysis = await analyzer.analyze(options.input);
+  } else {
+    analysis = await analyzer.analyzePackage();
+  }
+  /** Remember the user's preferences across runs. */
+  const lintActionsToAlwaysApply = new Set(options.edits || []);
+
+  const paths =
+      new Set([...analysis.getFeatures({kind: 'document'})].map((d) => d.url));
+  const watcher = new FilesystemChangeStream(
+      chokidar.watch([...paths], {persistent: true}));
+  for await (const changeBatch of watcher) {
```

Comment:
man, `for await` is wild.

So the intention is that this iterator never completes, until the user kills the process, correct? That would be worth a comment, just to make sure future readers don't follow through past this.

Reply:
I know right!? It's the best.

I added a comment mentioning that this is a forever loop.